components/product-info.js

```diff
@@ -9,23 +9,23 @@ import { pathToType } from "../lib/general";
 // If no url, then `{source.name} - {productId} - {lotId}`
 // We know if we have a lotId, then we have a productId
 export default function ProductInfo({
-  source = null,
+  sources = null,
```

Comment:
Give this an empty array as a default value (`sources = []`) so that we have no chance of line 17 causing a crash when it tries to use the length of a null property. You could argue to make this a non-optional property, but since conceivably someone else could call `ProductInfo` with a source that might not exist, I think keeping this optional works better.

Reply:
Thank you! I made it an empty array.